### PR TITLE
WIP: Start fixing Travis

### PR DESCRIPTION
**Pull request by C-Lodder**
Start fixing the travis issues. Don't merge yet as there's still some to go

### CODE REVIEW

**Comment by C-Lodder** [comment]
@mbabker All travis errors fixed apart from `'Joomla' is not defined  no-undef` which is the result of `JText`. Personally I'd suggest removing this and leaving it down to users to pass the pre-translated string into the alert.

@brianteeman thoughts?

File: src/js/alert/alert.js [hunk 1]
```diff
@@ -2,38 +2,45 @@
...
 
     /* Lifecycle, element appended to the DOM */
     connectedCallback() {
       this.setAttribute('role', 'alert');
-      this.classList.add("joomla-alert--show");
+      this.classList.add('joomla-alert--show');
 
       // Default to info
-      if (!this.type || ['info', 'warning', 'danger', 'success'].indexOf(this.type) === -1) {
+      if (['info', 'warning', 'danger', 'success'].indexOf(this.type) === -1) {
```
[the remark this answers is not in the document]
Comment (C-Lodder) [reply src/js/alert/alert.js]: I set it as the default here: https://github.com/joomla-projects/custom-elements/pull/104/files#diff-5a5c5f76b5305bd7907be3091c725edeR20

**Comment by wilsonge** [comment]
I've put a bunch of fixes into master just now so this PR is a bit simpler. Looking good. I agree about removing the Joomla.JText stuff. that can come from core.js for now.

**Comment by C-Lodder** [comment]
Ok cool. Can u merge this and I will do the JText stuff tonight after work?

**Comment by dgrammatiko** [comment]
@wilsonge you do realise that this PR broke the alerts, right?
screenshot 2018-10-20 at 10 04 07


https://codepen.io/dgrammatiko/pen/zmaNEr

**Pull request merged**